File: problems-adapt/0000-0100/0050_lone-element-among-triples/solution_bit_count.py
class Solution:
    def loneElementAmongTriples(self, nums: list[int]) -> int:
        result = 0
        for i in range(32):
            # Triples contribute 0 or 3 set bits at position i (a multiple of
            # three); the unique value contributes 0 or 1 — so count % 3 is
            # exactly bit i of the answer.
            count = 0
            for value in nums:
                count += (value >> i) & 1
            if count % 3 != 0:
                result |= 1 << i
        # Python integers are unbounded: reinterpret a pattern with bit 31 set
        # as the intended negative 32-bit value.
        if result >= 1 << 31:
            result -= 1 << 32
        return result

File: problems/0101-0200/0137_single-number-ii/solution_bit_count.java
class Solution {

    public int singleNumber(int[] nums) {
        int result = 0;
        for (int i = 0; i < 32; ++i) {
            // Triples contribute 0 or 3 set bits at position i (a multiple of
            // three); the unique value contributes 0 or 1 — so count % 3 is
            // exactly bit i of the answer.
            int count = 0;
            for (int value : nums) {
                count += (value >> i) & 1;
            }
            if (count % 3 != 0) {
                result |= 1 << i;
            }
        }
        return result;
    }
}
